mucommander-core/src/main/java/com/mucommander/ui/viewer/FileViewerPresenter.java

```diff
@@ -161,27 +175,50 @@ protected void show(AbstractFile file) throws IOException {
                     KeyStroke.getKeyStroke(KeyEvent.VK_M, ActionEvent.CTRL_MASK),
                     (e) -> {
                         boolean fullScreen = getFrame().isFullScreen();
-                        getFrame().setFullScreen(!fullScreen);
-                        fullScreenMenuItem.setSelected(!fullScreen);
-                    }
-            );
-            fullScreenMenuItem.setSelected(getFrame().isFullScreen());
+                        switchFullScreenMode(!fullScreen);
+                    });
+
+            String windowWidthValue = ViewerPreferences.WINDOW_WIDTH.getValue();
+            if (!"".equals(windowWidthValue)) {
+                int windowX = Integer.parseInt(ViewerPreferences.WINDOW_POSITION_X.getValue());
+                int windowY = Integer.parseInt(ViewerPreferences.WINDOW_POSITION_Y.getValue());
+                int windowWidth = Integer.parseInt(windowWidthValue);
+                int windowHeight = Integer.parseInt(ViewerPreferences.WINDOW_HEIGHT.getValue());
+                getFrame().setBounds(windowX, windowY, windowWidth, windowHeight);
+            } else {
+                getFrame().setDefaultBounds();
+            }
+
+            String showFullScreenValue = ViewerPreferences.SHOW_FULLSCREEN.getValue();
+            if (Boolean.TRUE.toString().equals(showFullScreenValue)) {
+                switchFullScreenMode(true);
+            }
             viewerMenu.add(fullScreenMenuItem);
 
             closeMenuItem = MenuToolkit.addMenuItem(viewerMenu,
                     Translator.get("file_viewer.close"),
                     menuItemMnemonicHelper,
                     KeyStroke.getKeyStroke(KeyEvent.VK_ESCAPE, 0),
                     (e) -> {
-                        fileViewer.close();
-                        getFrame().dispose();
+                        getFrame().dispatchEvent(new WindowEvent(getFrame(), WindowEvent.WINDOW_CLOSING));
```

Comment:
we can simplify it both here and in the viewer then with:
e -> getFrame().dispatchEvent(new WindowEvent(getFrame(), WindowEvent.WINDOW_CLOSING))